soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -120,6 +129,18 @@ static int nordicsemi_nrf53_init(const struct device *arg)
 	nrf_regulators_dcdcen_vddh_set(NRF_REGULATORS, true);
 #endif
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
+	DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_gpio_forwarder)
```

Comment:
Let's put this expression in a define to avoid repeating ourselves.

Reply:
Added `NRF_GPIO_FORWARDER_FOR_NRF5340_CPUAPP_ENABLED` to address this comment